dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi

```diff
@@ -0,0 +1,794 @@
+			};
+
+			pinctrl_adc0_in5: adc0_in5 {
+				pinmux = < MCHP_XEC_PINMUX(0205, MCHP_AF1) >;
+			};
+
+			pinctrl_adc0_in6: adc0_in6 {
+				pinmux = < MCHP_XEC_PINMUX(0206, MCHP_AF1) >;
+			};
+
+			pinctrl_adc0_in7: adc0_in7 {
+				pinmux = < MCHP_XEC_PINMUX(0207, MCHP_AF1) >;
+			};
+
+			pinctrl_adc0_vref2: adc0_vref2 {
+				pinmux = < MCHP_XEC_PINMUX(067, MCHP_AF1) >;
+			};
+
+			/* BBLED */
+			pinctrl_bbled0: bbled0 {
+				pinmux = < MCHP_XEC_PINMUX(0156, MCHP_AF1) >;
+			};
+
+			pinctrl_bbled1: bbled1 {
+				pinmux = < MCHP_XEC_PINMUX(0157, MCHP_AF1) >;
+			};
+
+			pinctrl_bbled1: bbled1 {
+				pinmux = < MCHP_XEC_PINMUX(0153, MCHP_AF1) >;
+			};
+
+			/* eSPI */
+			pinctrl_espi_alert: espi_alert {
+				pinmux = < MCHP_XEC_PINMUX(063, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_clk: espi_clk {
+				pinmux = < MCHP_XEC_PINMUX(065, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_cs: espi_cs {
+				pinmux = < MCHP_XEC_PINMUX(066, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_io0: espi_io0 {
+				pinmux = < MCHP_XEC_PINMUX(070, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_io1: espi_io1 {
+				pinmux = < MCHP_XEC_PINMUX(071, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_io2: espi_io2 {
+				pinmux = < MCHP_XEC_PINMUX(072, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_io3: espi_io3 {
+				pinmux = < MCHP_XEC_PINMUX(073, MCHP_AF1) >;
+			};
+
+			pinctrl_espi_reset: espi_reset {
+				pinmux = < MCHP_XEC_PINMUX(061, MCHP_AF1) >;
+			};
+
+			/* GPIO Pass Through */
+			pinctrl_gptp_in0: gptp_in0 {
+				pinmux = < MCHP_XEC_PINMUX(0224, MCHP_AF1) >;
+			};
+
+			pinctrl_gptp_out0: gptp_out0 {
+				pinmux = < MCHP_XEC_PINMUX(032, MCHP_AF2) >;
+			};
+
+			pinctrl_gptp_in1: gptp_in1 {
+				pinmux = < MCHP_XEC_PINMUX(016, MCHP_AF1) >;
+			};
+
+			pinctrl_gptp_out1: gptp_out1 {
+				pinmux = < MCHP_XEC_PINMUX(031, MCHP_AF2) >;
+			};
+
+			pinctrl_gptp_in2: gptp_in2 {
+				pinmux = < MCHP_XEC_PINMUX(014, MCHP_AF3) >;
+			};
+
+			pinctrl_gptp_out2: gptp_out2 {
+				pinmux = < MCHP_XEC_PINMUX(040, MCHP_AF1) >;
+			};
+
+			pinctrl_gptp_in3: gptp_in3 {
+				pinmux = < MCHP_XEC_PINMUX(0221, MCHP_AF2) >;
+			};
+
+			pinctrl_gptp_out3: gptp_out3 {
+				pinmux = < MCHP_XEC_PINMUX(0152, MCHP_AF3) >;
+			};
+
+			pinctrl_gptp_in5: gptp_in5 {
+				pinmux = < MCHP_XEC_PINMUX(017, MCHP_AF3) >;
+			};
+
+			pinctrl_gptp_out5: gptp_out5 {
+				pinmux = < MCHP_XEC_PINMUX(0125, MCHP_AF3) >;
+			};
+
+			pinctrl_gptp_in6: gptp_in6 {
+				pinmux = < MCHP_XEC_PINMUX(024, MCHP_AF2) >;
+			};
+
+			pinctrl_gptp_out6: gptp_out6 {
+				pinmux = < MCHP_XEC_PINMUX(0124, MCHP_AF4) >;
+			};
+
+			/* I2C port 0 */
+			pinctrl_i2c_scl0: i2c_scl0 {
+				pinmux = < MCHP_XEC_PINMUX(004, MCHP_AF1) >;
+			};
+
+			pinctrl_i2c_sda0: i2c_sda0 {
+				pinmux = < MCHP_XEC_PINMUX(003, MCHP_AF1) >;
+			};
+
+			/* I2C port 1 */
+			pinctrl_i2c_scl1: i2c_scl1 {
+				pinmux = < MCHP_XEC_PINMUX(0131, MCHP_AF1) >;
+			};
+
+			pinctrl_i2c_scl1_alt: i2c_scl1_alt {
+				pinmux = < MCHP_XEC_PINMUX(073, MCHP_AF2) >;
+			};
```

Comment:
I think signal names should be aligned with datasheet:

![image](https://user-images.githubusercontent.com/25011557/152313477-5004f095-9d2d-4d0a-b48b-027e96e8a67e.png)

datasheet uses "i2c01_scl", not "i2c_scl1".

Reply:
Could you confirm, this naming convention is okay:

```
        /* I2C port 1 */
	i2c01_scl: i2c01_scl {
		pinmux = < MCHP_XEC_PINMUX(0131, AF1) >;
	};

       i2c01_scl_alt: i2c01_scl_alt {
		pinmux = < MCHP_XEC_PINMUX(073, AF2) >;
	}
```;

Reply:
Please confirm, so that we can proceed to name pins as in the datasheet. Also note, we are not appending `default`.

Reply:
Looks good, but since you're not appending `default`, please, confirm the solution you expect to use for other states, such as `sleep`. It is important for nodes approach, since nodes can't be re-used for more than one state. It's a limitation people may not be aware since it is still not used by any driver, so it's better to think about. For example STM32 will use this approach:

```devicetree
/** autogenerated file **/

/* default state for i2c0 */
i2c0_scl_pa0 {
   pinmux = <STM32_PINMUX('A', 0, AF5);
};

i2c0_sda_pa1 {
    pinmux = <STM32_PINMUX('A', 1, AF5);
};

/* any pin is put into LP mode by using analog mode nodes,
 * this is how STM32 devices configure a pin for LP mode
 */
analog_pa0 {
    pinmux = <STM32_PINMUX('A', 0, ANALOG);
};

analog_pa1 {
    pinmux = <STM32_PINMUX('A', 1, ANALOG);
};

/** board file **/
&i2c0 {
    pinctrl-0 = <&i2c0_scl_pa0 &i2c0_sda_pa1>;
    pinctrl-0 = <&analog_pa0 &analog_pa1>;
    pinctrl-names = "default", "sleep";
};
```

Reply:
Thanks for the example.

Here is the updated naming convention we propose:


        ```
       /* I2C port 1 */
	i2c01_scl_gpio131: i2c01_scl_gpio131 {
		pinmux = < MCHP_XEC_PINMUX(0131, AF1) >;
	};

       i2c01_scl_alt_gpio073: i2c01_scl_alt_gpio073 {
		pinmux = < MCHP_XEC_PINMUX(073, AF2) >;
	};
```

For the later sleep mode support, we would add the following

```
	sleep_gpio131: sleep_gpio131 {
		pinmux = < MCHP_XEC_PINMUX(0131, AF0) >;
	};

       sleep_gpio073: sleep_gpio073 {
		pinmux = < MCHP_XEC_PINMUX(073, AF0) >;
	};
```

Reply:
Please confirm

Reply:
Naming LGTM.

Reply:
Updated naming convention as per the discussion.